packages/dds/matrix/src/matrix.ts

```diff
@@ -521,20 +523,34 @@ export class SharedMatrix<T = any>
 			SnapshotPath.cols,
 			this.cols.summarize(this.runtime, this.handle, serializer),
 		);
+		const cellsSnapshot = this.cells.snapshot();
+		const props: Record<string, number> = {
+			cellsSnapshotSize: cellsSnapshot.length,
+			rowCount: this.rowCount,
+			colCount: this.colCount,
+			setCellsCountFromLastSummary: this.setCellsCountFromLastSummary,
+		};
+		this.setCellsCountFromLastSummary = 0;
 		const artifactsToSummarize = [
-			this.cells.snapshot(),
+			cellsSnapshot,
 			this.pending.snapshot(),
 			this.setCellLwwToFwwPolicySwitchOpSeqNumber,
 		];
 
 		// Only need to store it in the snapshot if we have switched the policy already.
 		if (this.setCellLwwToFwwPolicySwitchOpSeqNumber > -1) {
-			artifactsToSummarize.push(this.cellLastWriteTracker.snapshot());
+			const trackerMatrixSnapshot = this.cellLastWriteTracker.snapshot();
+			artifactsToSummarize.push(trackerMatrixSnapshot);
+			props.trackerMatrixSnapshotSize = trackerMatrixSnapshot.length;
 		}
 		builder.addBlob(
 			SnapshotPath.cells,
 			serializer.stringify(artifactsToSummarize, this.handle),
 		);
+		this.logger.sendTelemetryEvent({
```

Comment:
I'm not sure this is applicable here but mention for completeness (and maybe you can see how to use it here).

You can avoid this event by leveraging ITelemetryContext object that is passed into summarizeCore() as a second argument. Summarizer process will automatically include it. That said, this only works as accumulation, i.e. that way we can record some stats about all shared matrixes that summarized on this path, but not individual DDSs.
For example, we can say that total number of cells across all such DDSs was X.

The benefit of using context object - there is only one single event per summary with all the properties collected from all summarized objects, i.e. efficiency in logging (and reduced frequency).

I do not see an easy way to leverage that mechanism here, unless we start using DDS Id to create unique records, which we should not do, due to IDs potentially having privacy concerns.

That said, this event you added might be too verbose. It all depends on a number of matrixes in a file and how often they get changed. I think that's fine for starters, but we will need to monitor it.

Reply:
Tried using this approach. Please take a look again.